server/resolvers/TestResultOperations/saveTestResultCommon.js

```diff
@@ -131,12 +131,8 @@ const assertTestResultIsValid = newTestResult => {
 
     const checkAssertionResult = assertionResult => {
         if (
-            !(
-                (assertionResult.passed === true &&
-                    !assertionResult.failedReason) ||
-                (assertionResult.passed === false &&
-                    !!assertionResult.failedReason)
-            )
+            assertionResult.passed === null ||
+            assertionResult.passed === undefined
```

Comment:
```suggestion
            !assertionResult.passed
```

Reply:
This looked good to me so I committed it but then the tests were failing. This made me realize that I can't do this because `assertionResult.passed === false` is a valid assertion result. Very easy to get tripped up on that 😅